### ISSUE TITLE
PRS: T-SQL OPENJSON WITH column definition clause causes parse failure

### ISSUE THREAD

**Issue by fank** [sqlfluff/sqlfluff#7547]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The T-SQL parser cannot handle `OPENJSON() WITH (...)` column definition syntax when used with `OUTER APPLY`. The `WITH (column_name type 'json_path')` clause is valid T-SQL syntax for defining the schema of the JSON output, but SQLFluff's tsql parser fails to recognize it, producing a PRS (unparsable section) error.

This cascades into false positives from other rules (RF01, RF02, etc.) because references within and after the unparsed section are not properly resolved.

### Expected Behaviour

The following query uses standard T-SQL `OPENJSON ... WITH` syntax and should parse without errors:

```sql
SELECT
    parsed_tag.tag_name
FROM
    OPENJSON(@json_input) AS main
    OUTER APPLY OPENJSON(main.value) WITH (
        tag_name NVARCHAR(128) N'$.tagName'
    ) AS parsed_tag
```

### Observed Behaviour

```
L:   5 | P:  35 |  PRS | Line 5, Position 35: Found unparsable section: "WITH
                       | (\n\t\ttag_name NVARCHAR(128) N'$.tagN..."
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
```

The parser fails at the `WITH (` token that begins the column definition clause of `OPENJSON`.

### How to reproduce

**test.sql:**
```sql
SELECT
    parsed_tag.tag_name
FROM
    OPENJSON(@json_input) AS main
    OUTER APPLY OPENJSON(main.value) WITH (
        tag_name NVARCHAR(128) N'$.tagName'
    ) AS parsed_tag
```

```bash
sqlfluff lint --dialect tsql test.sql
```

Output:
```
L:   5 | P:  35 |  PRS | Line 5, Position 35: Found unparsable section: "WITH
                       | (\n\t\ttag_name NVARCHAR(128) N'$.tagN..."
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
```

**More complex real-world pattern** that also fails — `OPENJSON WITH` combined with `EXCEPT`:

```sql
INSERT INTO dbo.tags (tag_name, created_by)
SELECT
    new_tag.tag_name
    ,@operator_id
FROM
    (
        SELECT
            parsed_tag.tag_name
        FROM
            OPENJSON(@json_input) AS main
            OUTER APPLY OPENJSON(main.value) WITH (
                tag_name NVARCHAR(128) N'$.tagName'
            ) AS parsed_tag

        EXCEPT

        SELECT
            tag_name
        FROM
            dbo.tags
        WHERE
            is_active = 1
    ) AS new_tag
```

### Dialect

tsql

### Version

sqlfluff 4.0.4, Python 3.10.12

### Configuration

```ini
[sqlfluff]
dialect = tsql
templater = raw
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
This has been recently fixed here: #7449, but not yet released. Pls check with the next release, and re-open if the issue still persist.
